src/App/Components/Common/NavProfile/Profile.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+import PropTypes from 'prop-types';
+import ReactImageFallback from 'react-image-fallback';
+import actions from '../../../../Redux/Actions/notifications';
+import defaultData from '../../../../configs/urls';
+import Triangle from '../Triangle';
+
+const { show, hide } = actions;
+
+export class Profile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isShown: false,
+      profile: {},
+      notifications: [],
+      notificationsBubble: ''
+    };
+  }
+
+  componentWillReceiveProps({
+    isShown, profile, configs, data
+  }) {
+    this.setState({ isShown });
+    if (!isEmpty({ ...profile, ...configs })) this.setState({ profile });
+    else this.setState({ profile: {} });
+    if (!isEmpty(data)) {
+      this.setState({ notifications: data });
+      if (data.length < 10) {
+        this.setState({ notificationsBubble: `${data.length}` });
+      } else this.setState({ notificationsBubble: '9+' });
+    }
+  }
+
+  showNotifications = () => {
+    const { state: { isShown } } = this;
+    const { show: showNotifications, hide: hideNotifications } = this.props;
+    if (isShown) {
+      return hideNotifications();
+    }
+    return showNotifications();
+  };
+
+  render() {
+    const {
+      state: {
+        profile, notifications, notificationsBubble, isShown
+      }
+    } = this;
+    return (
+      <React.Fragment>
+        <ul className="navbar-nav">
+          <li className="dropdown notify-container" id="notify-container">
+            <ReactImageFallback
+              src={profile ? profile.profileImage : ''}
+              fallbackImage={defaultData.defaultUserProfileImage}
+              alt="profile image"
+              className={
+                notifications.length > 0
+                  ? 'rounded-circle profile-img-notify'
+                  : 'rounded-circle profile-img'
+              }
+              onClick={this.showNotifications}
+            />
+            <Triangle direction="up" className={isShown ? 'show' : 'hide'} />
```

Comment:
@nkpremices what does the`Triangle` component do? Can its name be more descriptive?

Reply:
This component actually renders a triangle according to the direction that it is given. It's only for popups. Should I rename it something like `PopupTriangle`?

Reply:
@nkpremices the function of the component should inform its name. The fact that it renders a triangle is secondary to its purpose. 

Something like `popupAlert` would be more appropriate, I think.

Reply:
Noted @MemunaHaruna 
I'll do that